src/java/org/apache/cassandra/guardrails/Guardrails.java

```diff
@@ -109,6 +109,17 @@ what, formatSize(v), formatSize(t)))
                                                                     what, v, t))
                                .setMinNotifyIntervalInMs(TimeUnit.MINUTES.toMillis(30));
 
+    /**
+     * Guardrail on the number of dimensions of vector columns.
+     */
+    public static final Threshold vectorDimensions =
+            factory.threshold("vector_dimensions",
+                    () -> config.vector_dimensions_warn_threshold,
+                    () -> config.vector_dimensions_failure_threshold, // not needed so far
```

Comment:
isn't it the warn that's not needed so far, rather than the fail?

Reply:
The comment is wrong, I should have copy-pasted this from some other guardrail and missed to remove it. We add that comment when a threshold doesn't have a corresponding config property. In this case, both limits have an associated property, even if the warning is disabled by default. I've just removed the comment.